src/Core/Grid/Definition/Factory/GridDefinitionFactoryInterface.php

```diff
@@ -36,7 +36,7 @@
     /**
      * Create new grid definition.
      *
-     * @return DefinitionInterface
+     * @return GridDefinitionInterface
      */
     public function getDefinition();
```

Comment:
these `get*` methods for factories still looks weird for me and gives impression of singleton pattern. :/ i think it's more common for factories to have `create*` or `make*`, @mickaelandrieu  wdyt?

Reply:
`create` or `make` doesn't bring any value for people who don't have access (or don"t read: most common) to the docs.

A factory - by design - create something: what we need to know is "what?".

Reply:
Factory says what it creates, for example `FormFactoryInterface`, so we know it creates forms so methods like `create()`, `createNamed()` & etc does not need to repeat it, it wont create db connection or anything unrelated. On the other hand, we could have `createGrid()`?

Reply:
Yes, but look at how we use it behind a very long service name:

what we have is most of the time ``$this->get('prestashop.damn_very_long_name.factory')->create()``, I think it's better to have ``$this->get('prestashop.damn_very_long_name.factory')->getObject()`` because at the end of the day, it doesnt matter if we retrieve the object from factory or any creational pattern, what is important is to give meaningful information for developer who needs to manage the object :)